### ISSUE TITLE
Saving a model using a command does not terminate

### ISSUE THREAD

**Issue by francisjed** [adonisjs/adonis-framework#227]
I was trying to insert a user to users table and it was successful but the command does not terminate after `yield user.save()`. I think the iterator is in infinite loop.

Model:
```javascript
'use strict'

const Lucid = use('Lucid')

class User extends Lucid {
  
}

module.exports = User
```

Command:
```javascript
'use strict'

const Command = use('Command')
const User = use('App/Model/User')

class CreateUser extends Command {

  get signature() {
    return 'createUser {userName}'
  }

  get description() {
    return 'This will create and insert a user to DB'
  }

  * handle(args, options) {

    const user = new User()
    user.username = args.userName
    yield user.save() // Success, but wont terminate after this line
  }
}

module.exports = CreateUser
```

**Comment by thetutlage** [comment]
Database connections are long lived connections and you are required to close them manually. For `ace` commands you can do

```
const Database = use('Database')

* handle(args, options) {

    const user = new User()
    user.username = args.userName
    yield user.save() // Success, but wont terminate after this line
    Database.close()
  }
```

**Comment by francisjed** [comment]
@thetutlage Thanks for the fast reply! :)

**Comment by thetutlage** [comment]
Closing 😄

**Comment by lmj0011** [comment]
Would be nice to have a caveat or something in the docs related to this
http://adonisjs.com/docs/3.2/interactive-shell#_creating_your_first_command

**Comment by LordZombi** [comment]
I've got similar problem but with Mail. I'm sending notification emails using SMTP driver. Mails are sent but the command does not terminate. Any help please?

**Comment by 3dd13** [comment]
make sure all the external connections (e.g. database, redis) are all
explicitly closed

**Comment by LordZombi** [comment]
I do. I saw `Databse.close()` in doc example, so I close the Database like mentioned but Mail is what makes problems. I'm sending email after query - when I use Mail.fake() its OK but after Mail.send() command is not terminated I was looking for some "Mail.close()" but no success.

**Comment by 3dd13** [comment]
@LordZombi 

TL;DR
```
Mail.connection()._driverInstance.transporter.close()
```

## my troubleshooting steps

- able to repeat with latest adonis version 4.1
- then I look for places that has running connections, where I found SMTPPool
- then I look for places that I can close

included some source code line below for reference

- https://github.com/adonisjs/adonis-mail/blob/2494fd6d6c37198897cb5bdbb1b1b792278c85c2/src/Mail/index.js#L104
- https://github.com/adonisjs/adonis-mail/blob/develop/src/Mail/Sender.js#L24
- https://github.com/adonisjs/adonis-mail/blob/2494fd6d6c37198897cb5bdbb1b1b792278c85c2/src/Mail/Drivers/Smtp.js#L32

**Comment by LordZombi** [comment]
@3dd13 thanks for the workaround it works!

But I think Adonis can make nicer implementation :) This is kind oh hack when working with private variables (`_driverInstance`)

Still big thanks.
